File: Neetcode/SubSets2.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        op = list()
        nums.sort()
        def bt(i, res):
            if i >= len(nums):
                op.append(res.copy())
                return
            res.append(nums[i])
            bt(i+1, res)
            res.pop()
            while i+1 < len(nums) and nums[i+1] == nums[i]:
                i += 1
            bt(i+1, res)
        bt(0, [])
        return op

File: Neetcode/SubSets2.cpp
class Solution {
public:
    vector<vector<int>> out;
    vector<int> res;
    void bt(vector<int> nums, int i){
        if (i > nums.size()){
            return;
        }
        if (i == nums.size()){
            out.push_back(res);
            return;
        }
        res.push_back(nums[i]);
        bt(nums, i+1);
        while (i < nums.size() && nums[i] == res.back()){
            i += 1;
        }
        res.pop_back();
        bt(nums, i);
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        bt(nums, 0);
        return out;
    }
};
